packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -132,29 +136,71 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
     @inject(MonacoTextModelService)
     protected readonly textModelService: MonacoTextModelService;
 
+    private async openEx(commandId: string, resource: URI, columnOrOptions?: ViewColumn | TextDocumentShowOptions, viewType?: string): Promise<boolean | undefined> {
+        if (!resource) {
+            throw new Error(`${commandId} command requires at least URI argument.`);
+        }
+        if (!URI.isUri(resource)) {
+            throw new Error(`Invalid argument for ${commandId} command with URI argument. Found ${resource}`);
+        }
+
+        let options: TextDocumentShowOptions | undefined;
+        if (typeof columnOrOptions === 'number') {
+            options = {
+                viewColumn: columnOrOptions
+            };
+        } else if (columnOrOptions) {
+            options = {
+                ...columnOrOptions
+            };
+        }
+
+        const uri = new TheiaURI(resource);
+        const editorOptions = DocumentsMainImpl.toEditorOpenerOptions(this.shell, options);
+
+        let opener;
+        if (viewType) {
+            const lowerViewType = viewType.toLowerCase();
+            // Theia set custom editor id using 'custom-editor-' prefix
+            const customViewType = `custom-editor-${lowerViewType}`;
+            const openers = await this.openerService.getOpeners();
+            for (const opnr of openers) {
+                const idLowerCase = opnr.id.toLowerCase();
+                if (lowerViewType === idLowerCase || customViewType === idLowerCase) {
+                    opener = opnr;
+                    break;
+                }
+            }
+        } else {
+            opener = await this.openerService.getOpener(uri, editorOptions);
+        }
+
+        if (opener) {
+            opener.open(uri, editorOptions);
```

Comment:
You should await this `open` call. It could still throw, which would not be caught if we don't await it.